prime-router/src/main/kotlin/fhirengine/engine/QueueMessage.kt

```diff
@@ -72,6 +58,26 @@ abstract class Message {
     }
 }
 
+abstract class UniversalPipelineQueueMessage : QueueMessage() {
```

Comment:
One thing you could consider here is naming these abstract classes more around the functionality they provide.  For example, this abstract class captures functionality for downloading a blob from a `blobUrl` and `digest`.

To keep things more flexible moving forward, this could also become an interface.

Reply:
```kotlin
interface WithDownloadableReport {
    abstract val blobURL: String
    abstract val digest: String
    /**
     * Download the file associated with a RawSubmission message
     */
    fun downloadContent(): String {
        val blobContent = BlobAccess.downloadBlobAsByteArray(this.blobURL)
        val localDigest = BlobAccess.digestToString(BlobAccess.sha256Digest(blobContent))
        check(this.digest == localDigest) {
            "FHIR - Downloaded file does not match expected file\n${this.digest} | $localDigest"
        }
        return String(blobContent)
    }
}

Reply:
If I make it an interface though then it cannot inherit from the QueueMessage class.